src/v/kafka/server/handlers/details/security.h

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "kafka/protocol/schemata/create_acls_request.h"
+#include "security/acl.h"
+
+namespace kafka::details {
+
+/*
+ * Conversions throw acl_conversion_error and the exception message via (what())
+ * is generally what should be returned as the error message in kafka responses.
+ *
+ * Using an exception here eliminates the need to write c/go-style error
+ * handling for the large number of fields that need to be converted.
+ */
+struct acl_conversion_error : std::exception {
+    explicit acl_conversion_error(ss::sstring msg)
+      : msg{std::move(msg)} {}
+    const char* what() const noexcept final { return msg.c_str(); }
+    ss::sstring msg;
+};
+
+inline security::acl_principal to_acl_principal(const ss::sstring& principal) {
+    std::string_view view(principal);
+    if (unlikely(!view.starts_with("User:"))) {
+        throw acl_conversion_error(
+          fmt::format("Invalid principal name: {{{}}}", principal));
+    }
+    // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+    auto user = principal.substr(5);
```

Comment:
is it really limited to only 5 chars? 

@Mateoc - take note.

Reply:
no, the `User:` prefix is 5 chars

Reply:
OHHH! obvi. :D